### PR TITLE
[INTERNAL][HTML WB] Added drawing tools

### PR DESCRIPTION
**Pull request by AbdBarho**
these tools can be chosen from the menu, each tool has a mount() and unmount() functions to activate and deactive the tool respectively,
currently available tools are:
- dragDrawing: draw by holding down the mouse button and moving the mouse around
- freeDrawing: uses the drawing mode available in natively the rendering library Fabric.js, but converts the resulting shapes into ones that can be sent by the whiteboard to other users
- insertText: allows the users to insert text simply by clicking on a position and typing the wanted text
- selection: allows users to a select a shape and then resize it, move it, or detete it

all changes done in any of these tools are synchronized with other saros users

the ToolManager class is responsilbe for switching between the different tools.

### CODE REVIEW

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/services/toolmanager.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import LOGIC from '../constants/logic';
+import SVG from '../constants/SVG';
+import DragDrawing from '../tools/dragdrawing';
+import InsertText from '../tools/inserttext';
+import Selection from '../tools/selection';
+import FreeDrawing from '../tools/freedrawing';
+
+/**
+ * contatins all the tools that can be found in the tools menu,
```
Comment (m273d15) [inline de.fu_berlin.inf.dpp.whiteboard/frontend/src/services/toolmanager.js]: typo `contains`

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/services/toolmanager.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * any tool that is added here should have a mount() and unmount() functions which
```
Comment (m273d15) [inline de.fu_berlin.inf.dpp.whiteboard/frontend/src/services/toolmanager.js]: typo `function`

File: de.fu_berlin.inf.dpp.whiteboard/frontend/src/tools/freedrawing.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  /**
+   * converts the path in the event to an SXE path
+   * @param {Event} event
+   * fabricjs event which contains the path
+   */
+  overwritePath(event) {
+    // get needed parameters for the new path
+    let pathObj = event.path;
+    let { top, left } = pathObj;
+    let pathStr = pathObj.toSVG().match(/d="([^"]+)"/)[1];
```
Comment (m273d15) [inline de.fu_berlin.inf.dpp.whiteboard/frontend/src/tools/freedrawing.js]: Please add comment why this extraction via RegEx is necessary.

**Comment by AbdBarho** [comment]
@m273d15 I have updated the code and added more documentation

**Pull request merged**